### ISSUE TITLE
question about func "get_text_cond_grad"

### ISSUE THREAD

**Issue by sehwankim95** [YuxinWenRick/diffusion_memorization#9]
Hello, i have a question about the function "get_text_cond_grad".
I read the other issues about

```
single_prompt_embeds = prompt_embeds[[0], :, :].clone().detach()
single_prompt_embeds.requires_grad = True
dummy_prompt_embeds = prompt_embeds[[-1], :, :].clone()
input_prompt_embeds = torch.cat(
     [dummy_prompt_embeds.repeat(num_images_per_prompt, 1, 1),
      single_prompt_embeds.repeat(num_images_per_prompt, 1, 1),
      ]
)
```

so, I corrected above code to below.

```
single_prompt_embeds = prompt_embeds[[-1], :, :].clone().detach()
single_prompt_embeds.requires_grad = True
dummy_prompt_embeds = prompt_embeds[[0], :, :].clone()
input_prompt_embeds = torch.cat(
     [dummy_prompt_embeds.repeat(num_images_per_prompt, 1, 1),
      single_prompt_embeds.repeat(num_images_per_prompt, 1, 1),
      ]
)
```

However, after modifying the code and running it again, the results for the token-wise gradient changed.
For example, in the original paper, when computing the token-wise gradient for the prompt "Emma Watson to play Belle in Disney's <i>Beauty and the Beast</i>", tokens like Emma or Watson had higher gradient values, indicating stronger influence.
However, with the modified code, the gradient values consistently decrease from the SOS token to the EOS token, regardless of the prompt, and no longer reflect the impact of individual words.

I would appreciate it if you could let me know if there's anything I might have done wrong.

**Comment by sehwankim95** [comment]
(token_grads,) = torch.autograd.grad(loss, [single_prompt_embeds])
